dhcp-controller/api/v1alpha1/dhcpserver_types.go

```diff
@@ -22,18 +22,42 @@ import (
 
 // DHCPServerSpec defines the desired state of DHCPServer
 type DHCPServerSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Details of Servers
+	// +kubebuilder:validation:Required
+	Servers []Server `json:"servers,omitempty"`
+}
 
+type Server struct {
 	// refers to net-attach-def to be served
 	// +kubebuilder:validation:Required
 	NetworkName string `json:"networkName,omitempty"`
-	// refers to IP address to be configured at BindInterface port
+	// refers to IP address to bind interface to
 	// +kubebuilder:validation:Required
 	InterfaceIp string `json:"interfaceIp,omitempty"`
-	// configmap to be mounted as dnsmasq.conf
+	// refers to CIDR of server
+	// +kubebuilder:validation:Required
+	ServerCIDR CIDR `json:"cidr,omitempty"`
+	// refers to leasetime of IP
+	// +kubebuilder:validation:Required
+	LeaseTime string `json:"leaseTime,omitempty"`
+	// refers to vlan
+	VlanID string `json:"vlanid,omitempty"`
+}
+
+// CIDR defines CIDR of each network
+type CIDR struct {
+	// refers to start IP of range
+	// +kubebuilder:validation:Required
+	RangeStartIp string `json:"rangeStartIp,omitempty"`
```

Comment:
Can we make the range optional, and also have the user just give subnet cidr? For example, 10.1.0.0/16 or 192.168.1.0/24

Then we just use the entire subnet range. If they want to restrict it to a smaller allocation, pool, then they can give the start and end range. Also the cidr annotation is a *LITTLE* easier than asking for subnet mask - take a while to translate that the subnet mask for a /22 is 255.255.252.0 or a /20 is 255.255.240.0. And computing the range start and end can be awkward too for any subnetmask/cidr besides a /16, /24... since they don't fall in a clean .0 - .255 boundary.

Also the other main reason is - this puts us on par with Neutron and whereabouts. Both ask for at minimum a subnet and GW only, and in CIDR notation. RangeStart/End is optional, and they dont ask for a subnet mask.

Reply:
Done. Removed the subnet mask. Now the subnet CIDR is compulsory and if a start/end IP is given, it will override the defaults set by the subnet.

Reply:
remove the omitEmpty for the Cidr so it forces user to specify one. The yaml/JSON unmarshal will fail if empty

Reply:
Done